src/sidebar/media-embedder.js

```diff
@@ -301,7 +301,7 @@ function embedForLink(link) {
  *
  */
 function replaceLinkWithEmbed(link) {
-  if (link.href !== link.textContent) {
+  if (link.href !== link.textContent || decodeURI(link.href) !== link.textContent){
```

Comment:
This `if` statement will be true if `link.href` and `decodeURI(link.href)` are different, which will be true if there are any percent-encoded characters in the URL. I think you meant to use `&&` rather than `||` here to check that the link's text is not equal to either the encoded or unencoded versions of the URL.

Reply:
I'm sorry--I feel a bit confused. Does that mean I need a third conditional statement--i.e. checking decodeURI(link.href) against decodeURI (link.textContent) as well, or ...